keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,1229 @@
+the advancements in the compute vendor ecosystem.
+
+
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Users are requiring more varied workloads; the current set of available configurations
+for CPU, memory, and topology remain limited.  Additionally, the number of managers
+becoming internal to the Kubelet continues to increase, and we should find a more 
+dynamic and pluggable way of handling these resources.  Operating systems work by
+allowing drivers and pluggable resources, even to how cpu, memory, and devices are allocated.
+Kubernetes can be looked at as being the operating system of the cloud, and allowing
+specialty modules in order to address the use cases, rather than continuing to add
+complexity directly by continuing to modify the kubelet, will allow the greatest scope
+of abilities while stopping the continued increasing complexity within the core of
+the Kubelet.
+
+Users would like to be able to address the following use cases:
+
+* The ability to allow vendors to release vendor-specific managers for their hardware
+  and provide an interface while not having vendor-specific code within Kubelet.
+* Differentiate between types of cores and memory. <br>
+  Note - Dynamic resouce allocation does this with regular resources today.  We seek to
+  extend this ability.
+* Diffentiate between different configurations of cores and memory, for instance cores
+  designated as performance versus those designated as efficiency cores 
+* Have custom plugins to optimize for particular types of workloads.  These plugins
+  may be built for performance, power reduction, cpu savings, et cetera.  <br>
+  Note:  Currently, there are very limited sets of available topology policies.  Every
+  new policy must be approved and be lockstep with the Kuberenetes release process.
+* Be able to hot-plug and test new resource managers.  
+* Be able to remove some of the complexity with current setup and, over time, reduce
+  the amount of code contained within Kubelet.  Instead, build a library with specific
+  needs.
+* Have a faster path to desired changes, without potentially impacting the core of 
+  Kubernetes with every policy change. <br>
+  Note that current solutions have been cropping up to allow for resource management
+  outside the core Kubelet.  THese solutions require turning off the Kubelet functionality 
+  and overriding current Kubelet allocation.  We should provide a path otherwise.
+* Be able to get information on the pods on the node without having to contact the
+  API server, which may not have updated information. 
```

Comment:
Can you elaborate on this please? If API server doesn't have updated information how the scheduler is able to correctly schedule pods?